src/main/java/Advent/Day2.java

```diff
@@ -0,0 +1,80 @@
+package Advent;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Day2 {
+
+    public static void solve(List<String> input) {
+        System.out.println();
+        System.out.println("The answer of Day2 part1 is: " + solvePart1(input));
+        System.out.println();
+        System.out.println("The answer of Day2 part2 is: " + solvePart2(input));
+    }
+
+    public static int solvePart1(List<String> input) {
+        List<int[]> limits = limitsList(input);
+        char[] letters = lettersList(input);
+        List<String> codes = codesList(input);
+        int sumValidPasswords = 0;
+        for (int i = 0; i < limits.size(); i++) {
+            if (!passwordsValidation(limits.get(i), letters[i], codes.get(i)))
+                continue;
+            sumValidPasswords++;
+        }
+        return sumValidPasswords;
+    }
+
+    public static int solvePart2(List<String> input) {
+        List<int[]> limits = limitsList(input);
+        char[] letters = lettersList(input);
+        List<String> codes = codesList(input);
+        int sumValidPasswords = 0;
+        for (int i = 0; i < limits.size(); i++) {
+            if (!passwordsValidation2(limits.get(i), letters[i], codes.get(i))) {
+                continue;
+            }
+            sumValidPasswords++;
+        }
+        return sumValidPasswords;
+    }
+
+    private static boolean passwordsValidation(int[] limit, char letter, String code) {
+        int countLettersInCode = code.length() - code.replace(String.valueOf(letter), "").length();
+        return countLettersInCode >= limit[0] && countLettersInCode <= limit[1];
+    }
+
+    private static boolean passwordsValidation2(int[] limit, char letter, String code) {
+        char first = code.charAt(limit[0] - 1);
+        char second = code.charAt(limit[1] - 1);
+        return (Objects.equals(first, letter) && !Objects.equals(second, letter))
+                || (!Objects.equals(first, letter) && Objects.equals(second, letter));
+    }
+
+    private static List<int[]> limitsList(List<String> input) {
```

Comment:
Instead of a static function, you could populate parts of solutions like this...
In that case, you won't need to return n values- you will simply set them.
Now all this is very arbitrary and hard to understand.